internal/pkg/config/config.go

```diff
@@ -169,12 +225,23 @@ type rawSourceMatcher struct {
 }
 
 func (s *sourceMatcher) UnmarshalJSON(bytes []byte) error {
+	// Unknown config fields are not allowed.
+	rawMap := make(map[string]interface{})
+	if err := json.Unmarshal(bytes, &rawMap); err != nil {
+		return err
+	}
+	for label := range rawMap {
+		if !validSourceMatcherFields[label] {
+			return fmt.Errorf("%v is not a valid config field", label)
+		}
+	}
+
 	raw := rawSourceMatcher{}
 	if err := json.Unmarshal(bytes, &raw); err != nil {
 		return err
 	}
 
-	// validation: do not double-specify any attribute with literal and regexp
+	// Only one of literal and regexp field can be specified.
 	if raw.Package != nil && raw.PackageRE != nil {
 		return fmt.Errorf("expected only one of Package, PackageRE to be configured")
```

Comment:
For a future PR, if you wish: I think this error message should mention the fact that the code was trying to unmarshal a source definition.

Reply:
Added where the error is from to error message